spec/breadcrumbs/validator_spec.rb

```diff
@@ -0,0 +1,254 @@
+# encoding: utf-8
+require 'spec_helper'
+
+require 'bugsnag/breadcrumbs/types'
+require 'bugsnag/breadcrumbs/validator'
+
+RSpec.describe Bugsnag::Breadcrumbs::Validator do
+  before do
+    @automatic_breadcrumb_types = [
+      Bugsnag::Breadcrumbs::ERROR_BREADCRUMB_TYPE,
+      Bugsnag::Breadcrumbs::MANUAL_BREADCRUMB_TYPE,
+      Bugsnag::Breadcrumbs::NAVIGATION_BREADCRUMB_TYPE,
+      Bugsnag::Breadcrumbs::REQUEST_BREADCRUMB_TYPE,
+      Bugsnag::Breadcrumbs::PROCESS_BREADCRUMB_TYPE,
+      Bugsnag::Breadcrumbs::LOG_BREADCRUMB_TYPE,
+      Bugsnag::Breadcrumbs::USER_BREADCRUMB_TYPE,
+      Bugsnag::Breadcrumbs::STATE_BREADCRUMB_TYPE
+    ]
+
+    @breadcrumb_auto = false
+    @breadcrumb_message = "Valid message"
+    @breadcrumb_type = Bugsnag::Breadcrumbs::ERROR_BREADCRUMB_TYPE
+    @breadcrumb_meta_data = {}
+  end
+
+  describe "#Validate" do
+    it "does not 'ignore!' a valid breadcrumb" do
+      config = double
```

Comment:
It would be preferable to use `instance_double` instead of `double` as `instance_double` verifies that you don't stub something in an incorrect way. However, it would depend on the `Configuration` and `Breadcrumb` class changes first, so it might be good to organize the pull requests in a way that you can use `instance_double` here.

Reply:
I've made some progress towards this, although I still need to do a bunch of refactoring.  I'm happy for this PR to sit here for now while I make the configuration changes.